### ISSUE TITLE
Unexpected shadow label warning

### ISSUE THREAD

**Issue by notoria** [rust-lang/rust#81799]
This code produces the warning:
```Rust
fn main() {
    if true {
        'a: loop {
            break 'a;
        }
    } else {
        'a: loop {
            break 'a;
        }
    }
}
```
Playground [link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=57eb3928accd50dad69b9f90581d2f9e).

**Comment by ehuss** [comment]
I believe this is intended behavior, see #21633 and https://internals.rust-lang.org/t/psa-rejecting-duplicate-loop-labels/1833.

I think the warning should be converted to a future-compatible lint if the lang team really wants to keep it.

**Comment by notoria** [comment]
The label isn't in the same scope and there isn't a way to disable/`#[allow(...)]` this warning. Plus [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9ea7e4a26122b790d09a0455d71c544b) is accepted by the compiler.

**Comment by ehuss** [comment]
I think the intent is to make the entire function one scope, instead of using block scopes.  You are correct that there isn't a way to allow it, and that should be fixed (this warning was added pre-1.0, a lot has changed since then).

**Comment by ehuss** [comment]
Closing as a duplicate of #31745.
